proposed/http-client/http-client.md

```diff
@@ -21,9 +21,9 @@ implementations. This would make libraries more stable since the reduced number
 dependencies and the likelihood to get in version conflicts is reduced.
 
 The second goal is that all HTTP clients should follow the [Liskov substitutions principle][Liskov].
-This means that all clients should act the same when sending a request. By default a HTTP client
-should not follow redirect nor throw exceptions on HTTP responses with status 4xx or 5xx.
-
+This means that all clients should act the same when sending a request. A consuming
+library SHOULD expect a HTTP client to NOT follow redirects NOR throw exceptions based
+on HTTP response statuses.
```

Comment:
i am strictly -1 on a `bool $follow` argument. if anything, then flags, or an array of key-value that can be handled with eg. the symfony OptionsResolver.

but i am not happy to open this can of worms. there is a ton of things that could or could not be configured at runtime. it also implies that each client MUST be capable of doing those things (error checking, redirecting, ...whatever else we then want to add). reusable libraries would then be littered with configuration to each call where it should be configuration of the client behaviour. 

imo libraries should either use the low level client directly, or wrap it in something like the php-http PluginClient to make it handle redirects / throw exceptions (although those should be converted to domain exceptions actually, not http exceptions)...